src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py

```diff
@@ -350,3 +355,66 @@ def _update_offer_bid_trade(self) -> None:
         for area_uuid, area_result in self.flattened_area_core_stats_dict.items():
             self.bids_offers_trades[area_uuid] = {
                 k: area_result[k] for k in ("offers", "bids", "trades")}
+
+
+class CoefficientEndpointBuffer(SimulationEndpointBuffer):
+    """Calculate the endpoint results for the Coefficient based market."""
+
+    def _calculate_and_update_last_market_time_slot(self, area):
+        pass
+
+    def _populate_core_stats_and_sim_state(self, area: "Area"):
+        if area.uuid not in self.flattened_area_core_stats_dict:
+            self.flattened_area_core_stats_dict[area.uuid] = {}
+        if self.current_market_time_slot_str == "":
+            return
+
+        core_stats_dict = {"trades": []}
+        for trade in area.trades:
+            core_stats_dict["trades"].append(trade.serializable_dict())
+
+        if isinstance(area.strategy, SmartMeterStrategy):
+            core_stats_dict["smart_meter_profile_kWh"] = (
+                area.strategy.state.get_energy_at_market_slot(self.current_market_time_slot))
```

Comment:
I see that this method (`_populate_core_stats_and_sim_state()`) is mostly calling `area.strategy.state`. Do you think we could do something like `core_stats_dict = area.strategy.state.get_state()`? This might save us all the `if isinstance()` statements. What do you think?

Reply:
I agree, and I will go even further to name the state method as `get_current_state` in order to get the very latest state in order to remove the dependency of the `self.current_market_time_slot`, which IIRC is already included in some state classes (I distinctly recall it being used in the StorageState class). Give me some time, I'll prepare something, and thanks for the comment! At the first stage I will change only the coefficient-related code, but I will try to also refactor the market code at the same time if it does not take ages to do so.